[docs] Release notes for 2.21 (#21294)
yugabyte/yugabyte-db: docs/content/preview/releases/ybdb-releases/v2.21.md

---
title: What's new in the v2.21 preview release series
headerTitle: What's new in the v2.21 preview release series
linkTitle: v2.21 series (Preview)
description: Enhancements, changes, and resolved issues in the v2.21 preview release series.
image: /images/section_icons/quick_start/install.png
aliases:
  - /preview/releases/release-notes/latest-release/
  - /preview/releases/whats-new/latest-release
menu:
  preview_releases:
    identifier: v2.21
    parent: ybdb-releases
    weight: 2810
type: docs
---

{{< tip title="YugabyteDB Anywhere release notes have moved" >}}

> Starting with v2.16, the [release notes for YugabyteDB Anywhere](../../yba-releases/) have moved to [their own page](../../yba-releases/v2.21/).

Reviewer: use copy from 2.20 page

We're pleased to announce the tech preview of the new PostgreSQL emulation mode in the 2.21.0.0 release. This mode enables you to take advantage of many new improvements in both code compatibility and performance parity, thus making it even easier to lift and shift your applications from PostgreSQL to YugabyteDB. When this mode is turned on, YugabyteDB will use the [Read-Committed](../../../architecture/transactions/read-committed/) isolation mode, the [Wait-on-Conflict](../../../architecture/transactions/concurrency-control/#wait-on-conflict) concurrency mode for predictable P99 latencies, and the new Cost Based Optimizer that takes advantage of the distributed storage layer architecture and includes query pushdowns, LSM indexes, and [batched nested loop joins](../../../explore/ysql-language-features/join-strategies/#batched-nested-loop-join-bnl) to offer PostgreSQL like performance.

You can enable the emulation mode by passing the `enable_pg_parity_tech_preview` flag to [yugabyted](../../../reference/configuration/yugabyted/#yugabyted), when bringing up your cluster.

For example, from your YugabyteDB home directory, run the following command:

```sh

> ./bin/yugabyted start--enable_pg_parity_tech_preview

Reviewer: ```suggestion
./bin/yugabyted start --enable_pg_parity_tech_preview
```

```

**New YugabyteDB Kubernetes Operator**

A preliminary version of the completely rewritten YugabyteDB Kubernetes Operator is available in [Tech Preview](/preview/releases/versioning/#feature-availability). The new operator automates the deployment, scaling, and management of YugabyteDB clusters in Kubernetes environments. It streamlines database operations, reducing manual effort for developers and operators.

For more information, refer to the [YugabyteDB Kubernetes Operator](https://github.com/yugabyte/yugabyte-k8s-operator) GitHub project.

> **Changes in supported Operating Systems**

Reviewer: ```suggestion
**Changes to supported operating systems**
```

> Please note that YugabyteDB 2.21.0.0 and newer releases do not support v7 Linux versions (CentOS7, RHEL7, Oracle Enterprise Linux 7.x), Amazon Linux 2, and Ubuntu 18. If you're currently using one of these Linux versions, please upgrade to a supported OS version before installing YugabyteDB 2.21.0.0 version. Refer to [Operating system support](../../../reference/configuration/operating-systems/), for the complete list of supported Operating Systems.

Reviewer: ```suggestion
YugabyteDB 2.21.0.0 and newer releases do not support v7 Linux versions (CentOS7, Red Hat Enterprise Linux 7, Oracle Enterprise Linux 7.x), Amazon Linux 2, and Ubuntu 18. If you're currently using one of these Linux versions, upgrade to a supported OS version before installing YugabyteDB v2.21. Refer to [Operating system support](../../../reference/configuration/operating-systems/) for the complete list of supported operating systems.
```

### New features

* [New Kubernetes Operator](https://github.com/yugabyte/yugabyte-k8s-operator). Automated deployment and management of  clusters via the Kubernetes operator pattern. Includes support for YugabyteDB universes as a Kubernetes custom resource. Backup, upgrade, scale-out, scale-in, and more are possible on this Kubernetes custom resource. {{<badge/tp>}}

> * [Backup and restore support in yugabyted](../../../reference/configuration/yugabyted/). yugabyted CLI enabled support for taking database/keyspace backups. You now have the capability to upload backups to public clouds, including AWS and GCP. {{<badge/tp>}}

Reviewer: ```suggestion
* [Backup and restore support in yugabyted](../../../reference/configuration/yugabyted/). yugabyted now supports backup and restore of databases and keyspaces. You can also upload backups to public clouds, including AWS and GCP. {{<badge/tp>}}
```
